pkg/api/index.go

```diff
@@ -109,11 +109,19 @@ func (hs *HTTPServer) getAppLinks(c *models.ReqContext) ([]*dtos.NavLink, error)
 			}
 
 			if include.Type == "dashboard" && include.AddToNav {
-				link := &dtos.NavLink{
-					Url:  hs.Cfg.AppSubURL + "/dashboard/db/" + include.Slug,
-					Text: include.Name,
+				dashboards, err := hs.PluginManager.GetPluginDashboards(c.OrgId, plugin.Id)
```

Comment:
This can also be moved out of the loop, but then dashboards would be queried for every plugin (even those that don't contain any dashboards).